### PR TITLE
Delete entry

### PR DESCRIPTION
**Pull request by Drullan**
Also updated docs for this and previous new entry points.

### CODE REVIEW

**Comment by Drullan** [comment]
Seems like I accidentally merged in delete-user branch into this branch. I actually only added delete-entry route.

File: src/main/java/se/lth/cs/connect/modules/AccountSystem.java [hunk 1]
```diff
@@ -182,15 +184,65 @@ public static synchronized boolean createAccount(String email,
...
+        //find all collection user is owner of
+        //from those collection go back to all owners <----------------------|
+        //check if any of the owners have outgoing invites <-----------------|
+        //delete invites													 |
+        //delete collection and all relations								 |
+        //-------------------------------------------------------------------|
+        //these might become obsolete later if only one owner is allowed-----|
+       Database.query(db, new IClause[]{
+			MATCH.node().label("user").property("email").value(email)
+            	.relation().type("OWNER")
```
Comment (smeets) [inline src/main/java/se/lth/cs/connect/modules/AccountSystem.java:202]: Ok this is good. I feel like an event system is more and more beneficial for these kinds of clean ups.

File: src/main/java/se/lth/cs/connect/modules/AccountSystem.java [hunk 2]
```diff
@@ -182,15 +184,65 @@ public static synchronized boolean createAccount(String email,
...
+        Database.query(db, new IClause[]{
             MATCH.node(node).label("user").property("email").value(email),
             DO.DETACH_DELETE(node)
         });
 
+        removeCollectionWithNoUsers(db);
         return true;
     }
+    
+    private static void removeCollectionWithNoUsers(IDBAccess db){
```
Comment (smeets) [inline src/main/java/se/lth/cs/connect/modules/AccountSystem.java:225]: Any specific reason why you pass an `IDBAccess db` around?

File: src/main/java/se/lth/cs/connect/routes/Admin.java [hunk 3]
```diff
@@ -88,6 +90,52 @@ protected void setup(PippoSettings conf) {
...
+        	final JcQueryResult res = Database.query(rc.getLocal("db"), new IClause[]{
+        		MATCH.node().label("user").property("email").value(rc.getParameter("email").toString())
+        		.relation().type("OWNER").node(coll).label("collection"),
+        		RETURN.value(coll.property("name")).AS(cString)
+        	});
+        	
+            final List<String> found = res.resultOf(cString);
+            String[] colls = new String[found.size()];
+            found.toArray(colls);
+            rc.status(200).json().send(colls);
```
Comment (smeets) [inline src/main/java/se/lth/cs/connect/routes/Admin.java:110]: +1 for `RETURN.value(coll.property("name"))`

File: src/main/java/se/lth/cs/connect/routes/Admin.java [hunk 4]
```diff
@@ -88,6 +90,52 @@ protected void setup(PippoSettings conf) {
...
+        });
+        
+        // POST api.serp.se/v1/admin/delete-user
+        // email=...
+        POST("/delete-user", (rc) -> {
+        	if(rc.getParameter("email").isEmpty())
+        		throw new RequestException("must provide an user email parameter");
+
+        	AccountSystem.deleteAccount(rc.getParameter("email").toString(),rc.getLocal("db"));
+        	rc.getResponse().ok();
```
Comment (smeets) [inline src/main/java/se/lth/cs/connect/routes/Admin.java:121]: We should probably return some error if the request fails. Or is that taken care of in the deleteAccount function?

**Review by smeets — APPROVED** [review]

**Pull request merged**